### ISSUE TITLE
not able to play files from "your computer" (OSX)

### ISSUE THREAD

**Issue by bendschs** [CDrummond/cantata#1339]
Also with the latest version 2.3.2 i am not able to send local files to mpd with cantata. using OSX high sierra (10.13.6). Fails with the following message: "MPD reported the following error: failed to start playback."

i am using cantata together with https://github.com/ejurgensen/forked-daapd. In the newest version  
of cantata sending HTTP-streams is working properly. Sending local files to forked-daapd seems to be failing though.

cheers
bendsch

**Comment by CDrummond** [comment]
Have you ticked `Local file playback: [ ] Via in-built HTTP server` in the Collection settings page?

**Comment by bendschs** [comment]
yes, that one is ticked

**Comment by CDrummond** [comment]
Can you run Cantata from the commandline? Don't have access to a mac at the moment, but I believe you can run the executable from within the Cantata.app folder. If so, then please run with debug enabled - e.g. (paths may be incorrect):

```
./Cantata.app/Contents/MacOS/cantata --debug=mpd,http-server
```

This will spew lots of debug to the terminal window - I'm only interested in the debug that happens after you try to play a file. Please post that here.

-----

Also, can you play local files via the `Add Local Files` entry in the playqueue context menu?

**Comment by CDrummond** [comment]
OK, I have just tried - and this **does** work on macOS. Looks like this is an issue on your system, sorry.

**Comment by bendschs** [comment]
should the internal http-server be running all the time? is there any possibilty to check if its running?

**Comment by CDrummond** [comment]
No. Cantata's HTTP server **only** runs when there are local files in the playqueue. Also, and I think this is the root of the issue, Cantata will only serve files to connections from MPD. Cantata will check that the request is from MPDs IP, and it will also check that `User-Agent: Music Player Daemon` is in the request header. I would imagine forked-daap either does not set this, or sets it to another value. This would explain why it is working from macOS->MPD for me.

**Comment by CDrummond** [comment]
I've removed the user-agent checking, so this should be fixed when Cantata 2.3.3 is released.
